### DISCUSSION TITLE
umi建的网站在insight上做性能测试，TBT和Script Evaluation太慢，有解决办法吗？

### DISCUSSION THREAD

**Discussion by guijuzhejiang** [umijs/umi#12111] [General]
umi建的网站在insight上做性能测试，TBT和Script Evaluation太久，请教各位大侠有没有什么解决办法？
缩短 JavaScript 执行用时 2.4 秒
最大限度地减少主线程工作 3.0 秒
![image](https://github.com/umijs/umi/assets/57936895/e5d98146-d555-469d-9c89-cc6afbb24b22)
![image](https://github.com/umijs/umi/assets/57936895/9a836055-99e6-4589-917f-e1e81da10ed2)
![image](https://github.com/umijs/umi/assets/57936895/e90b51d1-2da5-4515-a68e-dafb783bc241)

**Comment by zxq08** [comment]
先尝试下umi自带的拆包配置吧，比如 codeSplitting。
可以通过 ANALYZE=1 pnpm max build  然后进本地8888分析下产物构成，可以考虑externals一些比较大的依赖包。
其实你可以先去performance分析下自己项目的卡点在哪里，比如请求长，异步请求多，页面结构复杂，可以从代码层面优化的。

**Reply by guijuzhejiang** [reply]
@zxq08 感谢回复，拆包这些优化我们都做了，本地的性能调试performance也看了，阻塞时间是232.32毫秒，如下图
![image](https://github.com/umijs/umi/assets/57936895/24266799-0ad9-4f91-9f6e-215aafd94c8b)

![image](https://github.com/umijs/umi/assets/57936895/e832fa7d-6f5d-4019-9cc1-edd9ee90616a)

![image](https://github.com/umijs/umi/assets/57936895/d65fc1d7-53e7-4c66-90ee-84d06f9114fa)
